### PR TITLE
v10: Optimize queries for rebuilding nucache

### PR DESCRIPTION
**Pull request by nzdev**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes 

### Description

Optimize queries for rebuilding nucache when using SQL Server / SQL Azure.
The rebuild queries time out when deleting existing nucache data on rebuild.
If all the cmsContentNu records will be deleted, truncates the table instead of running the deletes.
Added a todo with a suggestion on how to further optimize this in future.

How to test?
Rebuild nucache on SQL Server, SQL Azure, SQL CE.

### CODE REVIEW

File: src/Umbraco.PublishedCache.NuCache/Persistence/NuCacheContentRepository.cs [hunk 1]
```diff
@@ -206,19 +229,32 @@ private void RebuildMediaDbCache(IContentCacheDataSerializer serializer, int gro
...
             if (contentTypeIds == null || contentTypeIds.Count == 0)
             {
-                // must support SQL-CE
-                Database.Execute(
+                if (Database.DatabaseType is NPoco.DatabaseTypes.SqlServerCEDatabaseType)
+                {
+                    // must support SQL-CE
+                    Database.Execute(
                     @"DELETE FROM cmsContentNu
 WHERE cmsContentNu.nodeId IN (
```
Comment (p-m-j) [inline src/Umbraco.PublishedCache.NuCache/Persistence/NuCacheContentRepository.cs:237]: Could this utilize a join instead of a sub query?

**Comment by p-m-j** [comment]
I wonder if it's better to target v10 which really isn't far away.

**Comment by nzdev** [comment]
I think this is still useful for v9. I could open another pr after this on V10 that removes the SQLCE branches.

**Comment by ocdi** [comment]
Why did you close this pr? I suffer from this issue badly if I ever need to rebuild the cache. I am working around this issue by increasing the connection timeout, and it takes an hour to complete. I'm using a 2 core azure sql with 100,000 content nodes.

**Comment by nzdev** [comment]
@ocdi Mostly superseded by https://github.com/umbraco/Umbraco-CMS/pull/12785 which has been merged.

**Comment by nzdev** [comment]
I'll raise a new pr @ocdi

**Comment by nzdev** [comment]
Doone https://github.com/umbraco/Umbraco-CMS/pull/13501 @ocdi

**Pull request closed without merging**